src/Extensions/Diagnostics.HealthChecks/Internal/HttpEndpointHealthCheck.cs

```diff
@@ -59,6 +60,15 @@ protected override async Task<HealthCheckResult> CheckHealthInternalAsync(Health
 
 			HttpRequestMessage request = new HttpRequestMessage(Parameters.Method, m_uriToCheck);
 
+			foreach (KeyValuePair<string, IEnumerable<string>> pair in Parameters.Headers)
+			{
+				if (!request.Headers.TryAddWithoutValidation(pair.Key, pair.Value))
+				{
+					Logger.LogWarning(Tag.Create(), "Cannot add request header with name '{0}' value '{1}' for health check '{2}'.",
```

Comment:
Please add tests for the following cases:
key is null,
value is null,
value is an empty collection,
key is duplicated with differing values.

Reply:
key is null, value is null both throw compiler errors as the fields are marked non-nullable.
I added a test with empty collection and explicit test with multiple values in collection.
Key is duplicated with differing values isn't possible as the data is provided through Dictionaries and they throw on creation/when adding already existing keys.